client/alloc_watcher_test.go

```diff
@@ -196,3 +198,78 @@ func TestPrevAlloc_StreamAllocDir(t *testing.T) {
 		t.Fatalf("mode: %v", fi2.Mode())
 	}
 }
+
+// TestPrevAlloc_StreamAllocDir_Error asserts that errors encountered while
+// streaming a tar cause the migration to be cancelled and no files are written
+// (migrations are atomic).
+func TestPrevAlloc_StreamAllocDir_Error(t *testing.T) {
+	t.Parallel()
+	dest, err := ioutil.TempDir("", "nomadtest-")
+	if err != nil {
+		t.Fatalf("err: %v", err)
```

Comment:
Use assert test package- same for all other tests.

Reply:
With that said, the tests look great!

Reply:
I actually prefer this style as `assert` uses `t.Error` not `t.Fatal` so your test continues and usually panics due to the non-error value being nil.

...but maybe I'm the only one who gets confused by this? :confused: